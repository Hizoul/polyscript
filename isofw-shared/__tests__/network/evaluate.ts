import evaluateMeasurements from "isofw-shared/src/network/evaluate"
import testMeasurement from "isofw-shared/src/network/testMeasurement"

describe("measurements evaluation", () => {
  it("should calculate results", async () => {
    const measures = JSON.parse(testMeasurement)
    expect(await evaluateMeasurements(measures)).toMatchSnapshot("eval results")
  }, 5000)
})
