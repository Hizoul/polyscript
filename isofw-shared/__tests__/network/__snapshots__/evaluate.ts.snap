// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`measurements evaluation should calculate results: eval results 1`] = `
Object {
  "a": Object {
    "0": Object {
      "avgBytes": 745.3566666666667,
      "avgBytesPerMs": 76.47298221614227,
      "avgClientBytes": 182.83666666666667,
      "avgClientTime": 6.30048198312521,
      "avgClientToServer": 0.94,
      "avgMbytePerSecond": 0.0729303190385268,
      "avgNetworkTime": 6.30048198312521,
      "avgProcessTime": 9.746666666666666,
      "avgServerBytes": 562.52,
      "avgServerTime": 3.446184683541457,
      "avgServerToClient": 5.3966666666666665,
      "clientToServer": Array [],
      "fastestClientToServer": -1,
      "fastestNetworkTime": 1.6179990023374557,
      "fastestRoundTrip": 4,
      "fastestServerProcessTime": 4,
      "fastestServerToClient": 2,
      "medianClientToServer": 1,
      "medianNetworkTime": 4.881099998950958,
      "medianRoundTrip": 8,
      "medianServerProcessTime": 2.99550099670887,
      "medianServerToClient": 4,
      "networkTime": Array [],
      "roundTrip": Array [],
      "serverToClient": Array [],
      "servertime": Array [],
      "slowestClientToServer": 33,
      "slowestNetworkTime": 64.2369000017643,
      "slowestRoundTrip": 72,
      "slowestServerProcessTime": 72,
      "slowestServerToClient": 50,
      "total": 300,
      "totalBytes": 223607,
      "totalClientBytes": 54851,
      "totalClientTime": 1890.144594937563,
      "totalClientToServer": 282,
      "totalProcessTime": 2924,
      "totalServerBytes": 168756,
      "totalServerTime": 1033.855405062437,
      "totalServerToClient": 1619,
    },
  },
  "b": Object {
    "0": Object {
      "avgBytes": 691.61,
      "avgBytesPerMs": 61.89826968973747,
      "avgClientBytes": 163.37333333333333,
      "avgClientTime": 7.223768353809913,
      "avgClientToServer": 0.6566666666666666,
      "avgMbytePerSecond": 0.059030790033090086,
      "avgNetworkTime": 7.223768353809913,
      "avgProcessTime": 11.173333333333334,
      "avgServerBytes": 528.2366666666667,
      "avgServerTime": 3.9495649795234202,
      "avgServerToClient": 6.65,
      "clientToServer": Array [],
      "fastestClientToServer": -1,
      "fastestNetworkTime": 1.7695010006427765,
      "fastestRoundTrip": 4,
      "fastestServerProcessTime": 4,
      "fastestServerToClient": 2,
      "medianClientToServer": 1,
      "medianNetworkTime": 5.897799000144005,
      "medianRoundTrip": 10,
      "medianServerProcessTime": 3.969898998737335,
      "medianServerToClient": 6,
      "networkTime": Array [],
      "roundTrip": Array [],
      "serverToClient": Array [],
      "servertime": Array [],
      "slowestClientToServer": 8,
      "slowestNetworkTime": 47.31090000271797,
      "slowestRoundTrip": 53,
      "slowestServerProcessTime": 53,
      "slowestServerToClient": 46,
      "total": 300,
      "totalBytes": 207483,
      "totalClientBytes": 49012,
      "totalClientTime": 2167.130506142974,
      "totalClientToServer": 197,
      "totalProcessTime": 3352,
      "totalServerBytes": 158471,
      "totalServerTime": 1184.869493857026,
      "totalServerToClient": 1995,
    },
  },
  "c": Object {
    "0": Object {
      "avgBytes": 734.2033333333334,
      "avgBytesPerMs": 79.43058059862965,
      "avgClientBytes": 181.33,
      "avgClientTime": 6.22015940323472,
      "avgClientToServer": 0.6533333333333333,
      "avgMbytePerSecond": 0.07575090465414966,
      "avgNetworkTime": 6.22015940323472,
      "avgProcessTime": 9.243333333333334,
      "avgServerBytes": 552.8733333333333,
      "avgServerTime": 3.023173930098613,
      "avgServerToClient": 5.55,
      "clientToServer": Array [],
      "fastestClientToServer": -1,
      "fastestNetworkTime": 2.035400003194809,
      "fastestRoundTrip": 4,
      "fastestServerProcessTime": 4,
      "fastestServerToClient": 2,
      "medianClientToServer": 0,
      "medianNetworkTime": 5.0608000010252,
      "medianRoundTrip": 8,
      "medianServerProcessTime": 2.829700008034706,
      "medianServerToClient": 5,
      "networkTime": Array [],
      "roundTrip": Array [],
      "serverToClient": Array [],
      "servertime": Array [],
      "slowestClientToServer": 18,
      "slowestNetworkTime": 104.22570000588894,
      "slowestRoundTrip": 109,
      "slowestServerProcessTime": 109,
      "slowestServerToClient": 97,
      "total": 300,
      "totalBytes": 220261,
      "totalClientBytes": 54399,
      "totalClientTime": 1866.047820970416,
      "totalClientToServer": 196,
      "totalProcessTime": 2773,
      "totalServerBytes": 165862,
      "totalServerTime": 906.9521790295839,
      "totalServerToClient": 1665,
    },
  },
}
`;
